### PR TITLE
feat: Add option to print build statistics

### LINKED ISSUE #818: Print Lines of Code at the end of a build
### Feature request summary

Kraftkit should print the number of lines of code at the end of a build and also recommend enabling DCE if not enabled by default.

Support for LoC comes from the core and DCE can be detected by inspecting the kconfig file.

### Describe alternatives

_No response_

### Related architectures

None

### Related platforms

None

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by craciunoiuc**
### Prerequisite checklist

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes

These changes open the idea of not collapsing all output for every process. We could have custom log size for every step and the user just configures the default size.

GitHub-Fixes: #818

### CODE REVIEW

File: unikraft/app/application.go [hunk 1]
```diff
@@ -539,6 +542,45 @@ func (app application) Properclean(ctx context.Context, tc target.Target, mopts
...
+		if err := app.Prepare(
+			ctx,
+			tc,
+			append(
+				bopts.mopts,
+				make.WithProgressFunc(nil),
+			)...); err != nil {
+			return err
+		}
+	}
```
Comment (nderjung) [inline unikraft/app/application.go:574]: Is this necessary still?

File: cmd/kraft/build/build.go [hunk 2]
```diff
@@ -474,27 +475,52 @@ func (opts *Build) Run(cmd *cobra.Command, _ []string) error {
...
-					app.WithBuildLogFile(opts.SaveBuildLog),
-				)
-				if err != nil {
-					return fmt.Errorf("build failed: %w", err)
-				}
-				return nil
-			},
-		))
+		if opts.Stats {
+			// Print lines of code
```
Comment (nderjung) [inline cmd/kraft/build/build.go:479]: Does `make print-loc` perform a build or does it not just do something fancy after a build?

Comment (craciunoiuc) [reply cmd/kraft/build/build.go:479]: It invokes the build as a dependency, not sure if we can invoke it without actually building

File: unikraft/app/application.go (hunk 1 above)
Comment (craciunoiuc) [reply unikraft/app/application.go:574]: no idea 🤷

**Pull request closed without merging**